clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/services/EcsCloudMetricService.java

```diff
@@ -0,0 +1,249 @@
+import com.amazonaws.services.applicationautoscaling.model.DescribeScalingPoliciesResult;
+import com.amazonaws.services.applicationautoscaling.model.PutScalingPolicyRequest;
+import com.amazonaws.services.applicationautoscaling.model.PutScalingPolicyResult;
+import com.amazonaws.services.applicationautoscaling.model.ScalingPolicy;
+import com.amazonaws.services.applicationautoscaling.model.ServiceNamespace;
+import com.amazonaws.services.cloudwatch.AmazonCloudWatch;
+import com.amazonaws.services.cloudwatch.model.DeleteAlarmsRequest;
+import com.amazonaws.services.cloudwatch.model.DescribeAlarmsRequest;
+import com.amazonaws.services.cloudwatch.model.DescribeAlarmsResult;
+import com.amazonaws.services.cloudwatch.model.MetricAlarm;
+import com.amazonaws.services.cloudwatch.model.PutMetricAlarmRequest;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.cache.client.EcsCloudWatchAlarmCacheClient;
+import com.netflix.spinnaker.clouddriver.ecs.cache.model.EcsMetricAlarm;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Component
+public class EcsCloudMetricService {
+  @Autowired
+  EcsCloudWatchAlarmCacheClient metricAlarmCacheClient;
+  @Autowired
+  AccountCredentialsProvider accountCredentialsProvider;
+  @Autowired
+  AmazonClientProvider amazonClientProvider;
+
+  public void deleteMetrics(String serviceName, String account, String region) {
+    List<EcsMetricAlarm> metricAlarms = metricAlarmCacheClient.getMetricAlarms(serviceName, account, region);
+
+    if (metricAlarms.isEmpty()) {
+      return;
+    }
+
+    AmazonCredentials credentials = (AmazonCredentials) accountCredentialsProvider.getCredentials(account);
+    AmazonCloudWatch amazonCloudWatch = amazonClientProvider.getAmazonCloudWatch(account, credentials.getCredentialsProvider(), region);
+
+    amazonCloudWatch.deleteAlarms(new DeleteAlarmsRequest().withAlarmNames(metricAlarms.stream()
+      .map(MetricAlarm::getAlarmName)
+      .collect(Collectors.toSet())));
+
+    Set<String> resources = new HashSet<>();
+    // Stream and flatMap it? Couldn't figure out how.
+    for (MetricAlarm metricAlarm : metricAlarms) {
+      resources.addAll(buildResourceList(metricAlarm.getOKActions(), serviceName));
+      resources.addAll(buildResourceList(metricAlarm.getAlarmActions(), serviceName));
+      resources.addAll(buildResourceList(metricAlarm.getInsufficientDataActions(), serviceName));
+    }
```

Comment:
I think it's fine the way it is, but since you're interested...

```java
import org.junit.Test;

import java.util.ArrayList;
import java.util.Collection;
import java.util.HashSet;
import java.util.List;
import java.util.Set;
import java.util.stream.Collectors;

public class FooTest {

  @Test
  public void shouldDo() {
    List<MetricAlarm> metricAlarms = new ArrayList<>();
    metricAlarms.add(new MetricAlarm("hello"));
    metricAlarms.add(new MetricAlarm("world"));

    Set<String> resources = metricAlarms.stream()
      .map(metricAlarm -> {
        Set<String> r = new HashSet<>();
        r.addAll(metricAlarm.getOKActions());
        r.addAll(metricAlarm.getAlarmActions());
        r.addAll(metricAlarm.getInsufficientAlarms());
        return r;
      })
      .flatMap(Collection::stream)
      .collect(Collectors.toSet());

    assert resources.size() == 12;
  }

  private static class MetricAlarm {
    String name;

    MetricAlarm(String name) {
      this.name = name;
    }

    List<String> getOKActions() {
      return getActions("ok");
    }

    List<String> getAlarmActions() {
      return getActions("alarm");
    }

    List<String> getInsufficientAlarms() {
      return getActions("insufficient");
    }

    private List<String> getActions(String kind) {
      List<String> a = new ArrayList<>();
      a.add(name + ":" + kind + "1");
      a.add(name + ":" + kind + "2");
      return a;
    }
  }
}
```

Reply:
Thanks :)

Yeah it's not a whole lot better, but now I know.